rest-service/manager_rest/resources.py

```diff
@@ -317,8 +342,10 @@ def _save_file_locally(archive_file_name):
         else:
             if not request.data:
                 raise manager_exceptions.BadParametersError(
-                    'Missing application archive in request body or '
-                    '"blueprint_archive_url" in query parameters')
+                    'Missing {0} archive in request body or '
+                    '"{1}" in query parameters'.format(
+                        'application' if is_blueprint else 'snapshot',
```

Comment:
I guess no one will complain if you use kind variable instead of this if expression.